Src/Plugins/SalModernization/SalModernization.SharedStrings.txt

```diff
@@ -21,636 +21,632 @@
-$SEC105/606.ReturnValuePostAccessSAReadSARead=\[returnvalue:Post\(Access=\(SA_Read\s{0,4}|\s{0,4}SA_Read\)\)\]
-$SEC105/607.ReturnValuePostNullSAYes=\[returnvalue:Post\(Null=SA_Yes\)\]
-$SEC105/608.ReturnValuePostNullSANoDeref0=\[returnvalue:Post\(Null=SA_No,\s{0,4}Deref=0\)\]
-$SEC105/609.ReturnValuePostNullSANoDeref1=\[returnvalue:Post\(Null=SA_No,\s{0,4}Deref=1\)\]
-$SEC105/610.ReturnValuePostNullSAMaybeDeref2=\[returnvalue:Post\(Null=SA_Maybe,\s{0,4}Deref=2\)\]
-$SEC105/611.ReturnValuePostValidSAYes=\[returnvalue:Post\(Valid=SA_Yes\)\]
-$SEC105/612.ReturnValuePostWritableBytesConst1=\[returnvalue:Post\(WritableBytesConst=%1\)\]
-$SEC105/613.ReturnValuePostValidBytesConst1=\[returnvalue:Post\(ValidBytesConst=%1\)\]
-$SEC105/614.ReturnValuePostValidElementsConst1=\[returnvalue:Post\(ValidElementsConst=%1\)\]
-$SEC105/615.ReturnValuePostWritableElementsConst1=\[returnvalue:Post\(WritableElementsConst=%1\)\]
-$SEC105/616.ReturnValuePostNullTerminatedSANo=\[returnvalue:Post\(NullTerminated=SA_No\)\]
-$SEC105/617.ReturnValuePostDeref0ReturnValuePostValidSAYes=\[returnvalue:Post\(Deref=0\),\s{0,4}returnvalue:Post\(Valid=SA_Yes\)\]
-$SEC105/618.ReturnValueSAPostNullSAMaybe=\[returnvalue:\s{0,4}SA_Post\(Null=SA_Maybe\)\]
-$SEC105/619.ReturnValueSAPostDeref0NullSAYes=\[returnvalue:\s{0,4}SA_Post\(\s{0,4}Deref=\(0\),\s{0,4}Null=SA_Yes\s{0,4}\)\]
-$SEC105/620.ReturnValueSAPostNullSAMaybeValidElements1=\[returnvalue:\s{0,4}SA_Post\(Null=SA_Maybe,\s{0,4}ValidElements=%1\)\]
-$SEC105/621.ReturnValueSAPostDeref1NullSAMaybe=\[returnvalue:\s{0,4}SA_Post\(\s{0,4}Deref=\(1\),\s{0,4}Null=SA_Maybe\s{0,4}\)\]
-$SEC105/622.ReturnValueSAPostDeref0NullSANo=\[returnvalue:\s{0,4}SA_Post\(\s{0,4}Deref=\(0\),\s{0,4}Null=SA_No\s{0,4}\)\]
-$SEC105/623.ReturnValueSAPostMustCheckSAYes=\[returnvalue:\s{0,4}SA_Post\(MustCheck=SA_Yes\)\]
-$SEC105/624.ReturnValuePostAccessSAReadWrite=\[returnvalue:Post\(Access=SA_ReadWrite\)\]
-$SEC105/625.ReturnValuePostAccessSAReadWriteDeref1=\[returnvalue:Post\(Access=SA_ReadWrite,\s{0,4}Deref=1\)\]
-$SEC105/626.ReturnValuePostAccessSAWrite=\[returnvalue:Post\(Access=SA_Write\)\]
-$SEC105/627.ReturnValuePostAccessSAReadWriteSARead=\[returnvalue:Post\(Access=\(SA_ReadWrite\s{0,4}|\s{0,4}SA_Read\)\)\]
-$SEC105/628.ReturnValuePostValidSAMaybeDeref1=\[returnvalue:Post\(Valid=SA_Maybe,\s{0,4}Deref=1\)\]
-$SEC105/629.ReturnValueSAPostDeref1NullSAYesNullTerminatedSANo=\[returnvalue:\s{0,4}SA_Post\(\s{0,4}Deref=\(1\),\s{0,4}Null=SA_Yes,\s{0,4}NullTerminated=SA_No\s{0,4}\)\]
-$SEC105/630.ReturnValuePostAccessSAMaybe=\[returnvalue:Post\(Access=SA_Maybe\)\]
-$SEC105/631.ReturnValuePostAccessSANoAccess=\[returnvalue:Post\(Access=SA_NoAccess\)\]
-$SEC105/632.ReturnValuePostAccessSANoAccessSANoAccess=\[returnvalue:Post\(Access=\(SA_NoAccess\s{0,4}|\s{0,4}SA_NoAccess\)\)\]
-$SEC105/633.ReturnValuePostDeref1ValidSAYes=\[returnvalue:Post\(Deref='%1',\s{0,4}Valid=SA_Yes\)\]
\ No newline at end of file
+$SEC105/001/001.__allocator=[^\w_]__allocator[^\w_]
```

Comment:
>$SEC105/001/001 [](http://example.com/codeflow?start=0&length=15)

It isn't right to have two levels of number in an opaque id, no other analysis pattern does this and it's hard to read. Instead, drop the final id.

So this s/be $SEC105/001.allocator

Reply:
fixed.